core/src/main/java/com/graphhopper/routing/util/PriorityCode.java

```diff
@@ -24,14 +24,14 @@
  * @author Peter Karich
  */
 public enum PriorityCode {
-    WORST(0),
-    AVOID_AT_ALL_COSTS(1),
-    REACH_DEST(2),
-    AVOID_IF_POSSIBLE(3),
-    UNCHANGED(4),
-    PREFER(5),
-    VERY_NICE(6),
-    BEST(7);
+    WORST(5),
+    AVOID_AT_ALL_COSTS(6),
+    REACH_DEST(8),
+    AVOID_IF_POSSIBLE(9),
+    UNCHANGED(10),
+    PREFER(12),
+    VERY_NICE(13),
+    BEST(15);
```

Comment:
Maybe off-topic, but where do these names come from? `AVOID_AT_ALL_COSTS` sounds like its entirely forbidden, but it is still allowed and `WORST` is even more not allowed (but still not forbidden) :) Could we now also set it to `0`? Since we use four bits I would assume all values from 0 to 15 are possible?

Reply:
> Could we now also set it to 0? 

We should probably change/refactor the name and introduce new codes. But just setting them to different values will change routing behaviour where it is not intended.

Reply:
I could think about:

```
    EXCLUDE(0),
    VERY_BAD(3),
    BAD(5), // WORST
    REACH_DEST(6), // AVOID_AT_ALL_COSTS
    AVOID(8), // REACH_DEST
    AVOID_IF_POSSIBLE(9),
    UNCHANGED(10),
```

I'm unsure if we should find a name for every number. If values between are really required then they could be used directly?

Reply:
It seems the naming is taken from this OSM wiki: https://wiki.openstreetmap.org/wiki/Key:class:bicycle#Values

Reply:
> But just setting them to different values will change routing behaviour where it is not intended.

No, I did not mean to set different values. I was just wondering if it is now possible to set 0 as well.

> We should probably change/refactor the name and introduce new codes, yes

> Something like this?

Looks better yes.

Ok, or maybe use no names at all and just the value?

> I'm unsure if we should find a name for every number. If values between are really required then they could be used directly?

Yes, me neither :) Also 'best' and 'worst' sounds a bit confusing to me, I rather think of 'low' and 'high' priority.

> It seems the naming with 7 values is taken from this OSM wiki

Ok, I don't see much benefit in using these names, but we can also just leave it as is.

Reply:
Yes maybe just use the value directly? And maybe keep a constant for the max (15)? Not sure either way is fine for me, we can also leave this as it is, because it is not directly related to this issue.

Reply:
When thinking about this a bit then I would keep the enum values (as they also describe the intend a bit) although the current naming is indeed a bit confusing. Maybe @ratrun has an idea/opinion how to handle the increased possible values?

Reply:
I took the original naming from https://wiki.openstreetmap.org/wiki/DE:Key:class:bicycle. With 3 bits available this was kind of logical. Now that there is one more bit available I would like to use the most backward compatible solution.

Reply:
Ok, will try to come up with a better naming that does not use the old names so that not different values are used accidentially. What I like more about the new range is that we are able to exclude certain edges and that UNCHANGED is really keeping the value where before it was a "slight prefer".